Fix/ignore all remaining test failures when run under conda

tobiasdiez opened the pull request:
<!-- ^^^^^
Please provide a concise, informative and self-explanatory title.
Don't put issue numbers in there, do this in the PR body below.
For example, instead of "Fixes #1234" use "Introduce new method to calculate 1+1"
-->
<!-- Describe your changes here in detail -->

Add a `skip_conda` flag that skips the test when executed in an active conda environment. This is used to disable the remaining handful of tests that are still failing under linux in combination with https://github.com/sagemath/sage/pull/35593 (refs https://github.com/sagemath/sage/issues/35597).

<!-- Why is this change required? What problem does it solve? -->
<!-- If this PR resolves an open issue, please link to it here. For example "Fixes #12345". -->
<!-- If your change requires a documentation PR, please link it appropriately. -->

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Feel free to remove irrelevant items. -->

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

- #36411 (as it fixes some of the tests)
- https://github.com/sagemath/sage/pull/36468 (as it fixes the conda setup)

<!-- List all open PRs that this PR logically depends on
- #12345: short description why this is a dependency
- #34567: ...
-->

<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

mkoeppe commented on src/sage/doctest/parsing.py:
```diff
@@ -1203,6 +1204,9 @@ def check_and_clear_tag_counts():
                             ('not tested' in optional_tags)):
                         continue
 
+                    if ('skip_conda' in optional_tags and os.environ.get('CONDA_PREFIX', True)):
```
`True` is likely not the right default.

tobiasdiez replied:
Right, good catch!

mkoeppe commented on build/pkgs/ipython/distros/conda.txt:
```diff
@@ -1 +1,2 @@
-ipython
+ipython>=7.13.0,<8.9.0
+# ipython >= 8.9.0 requires prompt_toolkit too new for Sage
```
This change is obsolete with #36659

dimpase replied:
do we have a plan to switch to an up to date prompt_toolkit? I vaguely recall a discussion on this, but I can't find it.

mkoeppe replied:
#35251

tobiasdiez replied:
It's still required, see https://github.com/sagemath/sage/actions/runs/6804955526/job/18503541665?pr=36678.

mkoeppe replied:
That run didn't apply any CI fixes because https://github.com/sagemath/sage/actions/runs/6804955526/job/18503541665?pr=36678#step:3:7

orlitzky commented on src/sage/groups/matrix_gps/finitely_generated_gap.py:
```diff
@@ -120,7 +120,7 @@ def as_permutation_group(self, algorithm=None, seed=None):
             sage: P = G.as_permutation_group()
             sage: Psmaller = G.as_permutation_group(algorithm="smaller", seed=6)
-            sage: P == Psmaller
+            sage: P == Psmaller # skip_conda Fails for some reason when executed under conda
```
This is fixed in https://github.com/sagemath/sage/pull/36792 with a plausible explanation.

tobiasdiez replied:
Nice, I'll remove it here once https://github.com/sagemath/sage/pull/36792 has been merged - or we set that PR also to "blocker".

tobiasdiez committed:
Fix/ignore all remaining test failures when run under conda

tobiasdiez committed:
remove outdated test failures

tobiasdiez committed:
hopefully gap issues are fixed now

tobiasdiez committed:
Skip over timeouts in geometry/cone

Repreducible with sage -t --random-seed=53770599957939936954022527567356601238 src/sage/geometry/cone.py or sage -t --random-seed=294640319577531420678804037778614649656 src/sage/geometry/cone.py

tobiasdiez committed:
Skip over timeouts in fusion_ring

Not reproducible (at least not on linux)

tobiasdiez committed:
Remove outdated exceptions for older singular

https://github.com/sagemath/sage/issues/34851#issuecomment-1428888697

tobiasdiez committed:
remove outdated exception in lazy imports

tobiasdiez committed:
remove outdated exception for old threejs

tobiasdiez committed:
remove outdated exception for old pbori

orlitzky commented on src/sage/geometry/cone.py:
```diff
@@ -6364,9 +6364,8 @@ def random_cone(lattice=None, min_ambient_dim=0, max_ambient_dim=None,
-        ....:                 strictly_convex=False)
-        sage: K.is_strictly_convex()
+        sage: K = random_cone(min_ambient_dim=5, min_rays=2, strictly_convex=False) # skip_conda random timeouts
+        sage: K.is_strictly_convex() # skip_conda random timeouts
```
This one stands out to me because I wrote `random_cone()`. I don't think this example should time out. All we have to do here is generate a subspace (or even half-space) of dimension 5 or more, which is pretty easy to do by generating a bunch of random vectors and adding them to the list of generators one at a time. You have to be very unlucky for it to not be "instantaneous."

tobiasdiez replied:
See eg https://github.com/sagemath/sage/actions/runs/6827937663/job/18571052628#step:11:6663
